### PR TITLE
Locking tests upload

### PR DESCRIPTION
**Pull request by jvillafanez**
Added tests for uploads

### CODE REVIEW

**Comment by jvillafanez** [comment]
@nickvergessen @PVince81 @jnfrmarks 

FYI, there are some tests commented out because they're currently unreliable. It's probably better to remove them because it will be difficult to make sure a locked exception is thrown.

If everyone agrees, I'll remove those tests (leaving only chunked upload and zip download commented), remove unused functions, and merge into the "locking_tests" branch

File: lib/oc-tests/new_locking/test_uploadFilePlusAction.py [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+    return os.path.getsize(localpath) == size
+
+def check_zip_contents(localpath, content_list):
+    checked_result = False
+    with zipfile.ZipFile(localpath, 'r') as myzip:
+        checked_result = myzip.namelist() == content_list
+    return checked_result
+
+def parse_worker_number(worker_name):
+    match = re.search(r'(\d+)$', worker_name)
```
Comment (nickvergessen) [inline lib/oc-tests/new_locking/test_uploadFilePlusAction.py:44]: Nop https://github.com/owncloud/smashbox/pull/90/files

Comment (nickvergessen) [inline lib/oc-tests/new_locking/test_uploadFilePlusAction.py:44]: or is this not the full path?

File: lib/oc-tests/new_locking/test_uploadFilePlusAction.py [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+        return int(match.group())
+    else:
+        return None
+
+
+testsets = [
+        { 'action_method': 'put_file_contents',
+          'action_args': ('/folder/new_bigfile.dat', '123'*50),
+          'action_kwargs': {},
+          'accounts': sconf.oc_number_test_users,
```
Comment (nickvergessen) [inline lib/oc-tests/new_locking/test_uploadFilePlusAction.py:55]: what is `sconf.oc_number_test_users`

File: lib/oc-tests/new_locking/test_uploadFilePlusAction.py (hunk 1 above)
Comment (jvillafanez) [inline lib/oc-tests/new_locking/test_uploadFilePlusAction.py:44]: I'll make the change after merging to the branch. Download and overwrite tests will need the same change, so I'll do it at the same time

File: lib/oc-tests/new_locking/test_uploadFilePlusAction.py (hunk 2 above)
Comment (jvillafanez) [inline lib/oc-tests/new_locking/test_uploadFilePlusAction.py:55]: It's the number of user accounts as defined in the smashbox.conf file.

Comment (nickvergessen) [inline lib/oc-tests/new_locking/test_uploadFilePlusAction.py:55]: Well there is no need to specify this in the testcase then, just use `config.oc_number_test_users` in the test directly?

File: lib/oc-tests/new_locking/test_uploadFilePlusAction.py (hunk 1 above)
Comment (nickvergessen) [inline lib/oc-tests/new_locking/test_uploadFilePlusAction.py:44]: Okay, nevermind.
Seems like `parse_worker_number(reflection.getProcessName())` is what we want to have in the end.

File: lib/oc-tests/new_locking/test_uploadFilePlusAction.py (hunk 2 above)
Comment (jvillafanez) [inline lib/oc-tests/new_locking/test_uploadFilePlusAction.py:55]: We might need to change it or setting lower and upper limits (use more than 1 account, for example) on a per-test basis. I prefer to keep the versatility.

**Comment by jvillafanez** [comment]
One last call before merging into the "locking_test" branch. If there isn't any more suggested change I'll merge by the end of today

**Pull request merged**